core/src/Platforms/iOS/PlatformProxy.cs

```diff
@@ -0,0 +1,79 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using UIKit;
+
+namespace Microsoft.Identity.Core
+{
+    /// <summary>
+    /// Platform / OS specific logic.  No library (ADAL / MSAL) specific code should go in here. 
+    /// </summary>
+    internal class PlatformProxy : IPlatformProxy
```

Comment:
Could we start naming the classes more specifically?  The intellisense/build break navigation logic gets all confused with 5+ implementations of exactly the same class based on the configuration.  

So, could we call this IosPlatformProxy as a class (and Net45PlatformProxy, etc for others).  Then we can have a single factory class called PlatformProxyFactory() that returns an IPlatformProxy and within that factory we have the #ifdef code to look for the platform we're building for and return the right one.  I think this would be really helpful for our own code navigation and debugging...

Reply:
+1: imo a good idea 

---
In reply to: [221390351](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1284#discussion_r221390351) [](ancestors = 221390351)